stix_shifter_modules/nozomi/stix_translation/query_constructor.py

```diff
@@ -0,0 +1,549 @@
+        """
+        parse mapped fields into boolean expression
+        :param formatted_value: str
+        :param mapped_fields_array: list
+        :param mapped_field_type:str
+        :param expression: expression object
+        :return: list
+        """
+        comparator = self._lookup_comparison_operator(expression.comparator)
+
+        if expression.negated:
+            if expression.comparator == ComparisonComparators.Equal:
+                comparator = self._lookup_comparison_operator(ComparisonComparators.NotEqual)
+            else:
+                comparator = f'!{comparator}'
+
+        if expression.comparator in \
+                (ComparisonComparators.Like, ComparisonComparators.In, ComparisonComparators.IsSubSet):
+            comparator = f' {comparator} '
+
+        if expression.object_path == 'x-ibm-finding:finding_type':
+            formatted_value, comparator = self._handle_threat_name(formatted_value, comparator)
+
+        if mapped_field_type == 'subset':
+            mapped_fields_array = self.config_map['subset_supported_fields']
+
+        if isinstance(formatted_value, list):
+            formatted_value_list = formatted_value
+        else:
+            formatted_value_list = [formatted_value]
+
+        comparison_list = []
+
+        for value in formatted_value_list:
+            comparison_string_new = 'where '
+            for index, field_name in enumerate(mapped_fields_array):
+                if index > 0 and comparison_string_new != 'where ':
+                    comparison_string_new += ' OR '
+
+                if mapped_field_type == 'properties' and 'properties' in field_name:
+                    like_operator = self._lookup_comparison_operator(ComparisonComparators.Like)
+                    if '!' in comparator:
+                        like_operator = f'!{like_operator}'
+                    comparison_string_new += f'{field_name} {like_operator} {value}'
+                else:
+                    comparison_string = f'{field_name}{comparator}{value}'
+                    if len(comparison_string_new + comparison_string) > MAX_QUERY_LENGTH:
+                        if comparison_string_new[-4:] == ' OR ':
+                            comparison_string_new = comparison_string_new[:-4]
+                        comparison_list.append(comparison_string_new)
+                        comparison_string_new = 'where '
+                    comparison_string_new += comparison_string
+            comparison_list.append(comparison_string_new)
+
+        return comparison_list
+
+    @staticmethod
+    def check_common_timestamp(query_01, query_02):
+        """
+        Check the queries contains same timestamp
+        :param query_01: str
+        :param query_02: str
+        :return query_01_without_timestamp str, query_02_without_timestamp str, timestamp str
+        """
+        # Last 81 characters in the query string contains timestamp value
+        if query_01[-81:] == query_02[-81:]:
```

Comment:
This seems really prone to breaking. Wouldn't it be better to use comparison checks rather than a hardcoded character count?

Reply:
Hardcoded value removed, Updated as per the suggestion.